broker/src/asterisk/asterisk_channel_srv.erl

```diff
@@ -108,7 +121,8 @@ handle_cast(regenerate_config, State = #state{config_job_state = JobState}) ->
         ami_client:sip_reload(),
         gen_server:cast(?MODULE, {set_channels, ChannelIndex, RegistryIndex})
       end),
-      State#state{config_job_state = working};
+      % reset the count of registration failures
+      State#state{config_job_state = working, registration_failures = dict:new()};
```

Comment:
That would be ideal, but implies a lot more state needs to be kept somewhere. `updated_at` for each channel at the very least. Implementation shouldn't be very difficult.